client/connection/manager.go

```diff
@@ -2,44 +2,57 @@ package connection
 
 import (
 	"errors"
+	log "github.com/cihub/seelog"
 	"github.com/mysterium/node/communication"
 	"github.com/mysterium/node/identity"
 	"github.com/mysterium/node/openvpn"
 	"github.com/mysterium/node/openvpn/middlewares/client/bytescount"
 	"github.com/mysterium/node/server"
 	"github.com/mysterium/node/service_discovery/dto"
 	"github.com/mysterium/node/session"
+	"github.com/mysterium/node/utils"
 )
 
+const managerLogPrefix = "[connection-manager] "
+
+const channelClosed = openvpn.State("")
+
 var (
 	// ErrNoConnection error indicates that action applied to manager expects active connection (i.e. disconnect)
 	ErrNoConnection = errors.New("no connection exists")
 	// ErrAlreadyExists error indicates that aciton applieto to manager expects no active connection (i.e. connect)
 	ErrAlreadyExists = errors.New("connection already exists")
+	// ErrConnectionCancelled indicates that connection in progress was cancelled by request of api user
+	ErrConnectionCancelled = errors.New("connection was cancelled")
+	// ErrOpenvpnProcessDied indicates that Connect method didn't reach "Connected" phase due to openvpn error
+	ErrOpenvpnProcessDied = errors.New("openvpn process died")
 )
 
 type connectionManager struct {
 	//these are passed on creation
-	mysteriumClient      server.Client
-	newDialogEstablisher DialogEstablisherCreator
-	newVpnClient         VpnClientCreator
-	statsKeeper          bytescount.SessionStatsKeeper
+	mysteriumClient server.Client
+	newDialog       DialogCreator
+	newVpnClient    VpnClientCreator
+	statsKeeper     bytescount.SessionStatsKeeper
 	//these are populated by Connect at runtime
-	status        ConnectionStatus
-	dialog        communication.Dialog
-	openvpnClient openvpn.Client
-	sessionID     session.SessionID
+	status      ConnectionStatus
+	closeAction func()
+}
+
+func warnOnClose() {
+	log.Warn(managerLogPrefix, "WARNING! Trying to close when there is nothing to close. Possible bug or race condition")
```

Comment:
We don't need to start each warning with "WARNING!" - `log.Warn` is responsible to show that this is a warning :D

Reply:
Removed